src/appleseed/renderer/modeling/bsdf/hairbsdf.cpp

```diff
@@ -582,15 +582,15 @@ namespace
                     values->m_precomputed.m_v,
                     sample_M);
             }
-            if (p == 1)
+            else if (p == 1)
             {
                 sin_theta_i = sample_longitudinal(
                     std::sin(theta_o_TT),
                     std::cos(theta_o_TT),
                     values->m_precomputed.m_v * 0.25f,
                     sample_M);
             }
-            if (p == 2)
+            else if (p == 2)
```

Comment:
You can just do `else` here. Like this you don't need to do `sin_thetat_i = 0.0f` and `sin_theta_i` will always be initialized.

Reply:
Thanks for your commenting, have resolved.